### PR TITLE
Refactor baseline and api scan scripts

### PR DESCRIPTION
**Pull request by psiinon**
The new zap_common.py script has an underscore in its name as you cant
import files with a minus in :( Dont really want to change the other
file names as that will break things for people.
I'm sure the refactoring could be better, but its a start ;)
Feel free to comment but please dont check in yet (in the unlikely event
they are approved first time;) - I'm running them against the Mozilla
sites we use the existing scripts with as a sanity check.

### CODE REVIEW

File: build/docker/zap-api-scan.py [hunk 1]
```diff
@@ -52,21 +52,19 @@
 import os
 import os.path
 import re
```
Comment (thc202) [inline build/docker/zap-api-scan.py]: Could be removed,  no longer used.

File: build/docker/zap-api-scan.py [hunk 2]
```diff
@@ -52,21 +52,19 @@
...
-import socket
 import subprocess
 import sys
 import time
-import traceback
 import urllib2
 from datetime import datetime
 from random import randint
```
Comment (thc202) [inline build/docker/zap-api-scan.py]: Could be removed,  no longer used.

File: build/docker/zap-baseline.py [hunk 3]
```diff
@@ -43,26 +43,21 @@
 # at the end of each line.
 
 import getopt
-import json
 import logging
 import os
 import os.path
 import re
```
Comment (thc202) [inline build/docker/zap-baseline.py]: Could be removed, no longer used.

File: build/docker/zap-baseline.py [hunk 4]
```diff
@@ -311,79 +232,47 @@ def main(argv):
...
         # Not running in docker, so start one
-        try:
-            logging.debug('Pulling ZAP Weekly Docker image')
-            ls_output = subprocess.check_output(['docker', 'pull', 'owasp/zap2docker-weekly'])
-        except OSError:
-            logging.warning('Failed to run docker - is it on your path?')
-            sys.exit(3)
+        mount_dir = ''
+        if context_file:
+            mount_dir =  os.path.dirname(os.path.abspath(context_file))
```
Comment (thc202) [inline build/docker/zap-baseline.py]: There's an extra space after the equal (PEP8, E222).

File: build/docker/zap-baseline.py [hunk 5]
```diff
@@ -419,39 +308,18 @@ def main(argv):
...
             while ((datetime.now() - start_scan).seconds < delay):
                 time.sleep(5)
                 logging.debug('Delay passive scan check ' + str(delay - (datetime.now() - start_scan).seconds) + ' seconds')
-        rtc = zap.pscan.records_to_scan
-        logging.debug('Records to scan...')
-        while (int(zap.pscan.records_to_scan) > 0):
-            logging.debug('Records to passive scan : ' + zap.pscan.records_to_scan)
-            time.sleep(2)
-        logging.debug('Passive scanning complete')
+
```
Comment (thc202) [inline build/docker/zap-baseline.py]: The white space chars could be removed (PEP8, W293).

File: build/docker/zap_common.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        print (action + '-IN_PROGRESS: ' + alert_list[0].get('alert') + ' [' + id + '] x ' + str(len(alert_list)) + ' ' + user_msg)
+        if in_progress_issues[id]["link"]:
+            print ('\tProgress link: ' + in_progress_issues[id]["link"])
+    else:
+        print (action + '-NEW: ' + alert_list[0].get('alert') + ' [' + id + '] x ' + str(len(alert_list)) + ' ' + user_msg)
+    if detailed_output:
+        # Show (up to) first 5 urls
+        for alert in alert_list[0:5]:
+            print ('\t' + alert.get('url'))
+
```
Comment (thc202) [inline build/docker/zap_common.py]: Should have another empty line (PEP8, E302).

File: build/docker/zap_common.py [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+
+def inc_fail_rules(config_dict, key, inc_extra):
+    return config_dict.has_key(key) and config_dict[key] == 'FAIL'
+
+
+def dump_log_file(cid):
+    traceback.print_exc()
+    # Unexpected issue - dump the zap.log file
+    if running_in_docker:
```
Comment (thc202) [inline build/docker/zap_common.py]: Should call the function.

File: build/docker/zap_common.py [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+
+
+def zap_wait_for_passive_scan(zap):
+    rtc = zap.pscan.records_to_scan
+    logging.debug('Records to scan...')
+    while (int(zap.pscan.records_to_scan) > 0):
+        logging.debug('Records to passive scan : ' + zap.pscan.records_to_scan)
+        time.sleep(2)
+    logging.debug('Passive scanning complete')
+
```
Comment (thc202) [inline build/docker/zap_common.py]: Should have another empty line (PEP8, E302).

File: build/docker/zap_common.py [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+                # Ignore all info alerts - some of them may have been downgraded by security annotations
+                continue
+            if (not alert_dict.has_key(plugin_id)):
+                alert_dict[plugin_id] = []
+            alert_dict[plugin_id].append(alert)
+        st += pg
+        alerts = zap.core.alerts(start=st, count=pg)
+    logging.debug('Total number of alerts: ' + str(alert_count))
+    return alert_dict
+
```
Comment (thc202) [inline build/docker/zap_common.py]: Empty line should be removed (PEP8, W391).

**Comment by psiinon** [comment]
Thanks - hopefully fixed all of those

File: build/docker/zap-baseline.py (hunk 4 above)
Comment (thc202) [inline build/docker/zap-baseline.py]: This one was not addressed.

Comment (psiinon) [inline build/docker/zap-baseline.py]: Good spot :) Fixed now

**Comment by thc202** [comment]
Thanks!

**Pull request merged**